### PR TITLE
📝 Update data log

### PR DESCRIPTION
**Pull request by StatsRhian**
I've made a start on the data log

### CODE REVIEW

File: data_extraction/data_log.qmd [hunk 1]
```diff
@@ -5,6 +5,28 @@ order: 250
 
 This page contains information about changes to the data underpinning the NHP model. If there are no changes logged for a particular version (such as 1.0) then the changes were only to the model code, and not to the underlying data.
 
+## Version 3.0
+
+Date updated: 13/11/2024
+
+As part of migrating our data processing from SQL to databricks, we have a made a number of changes to the data.
```
[the remark this answers is not in the document]
Comment (StatsRhian) [reply data_extraction/data_log.qmd]: Great shout. Added. 🙏🏻

File: data_extraction/data_log.qmd [hunk 2]
```diff
@@ -5,6 +5,28 @@ order: 250
...
+
+Inpatient data:
+  - Include records where `patientid` is `NULL`
+  - Include readmission within 28 days for well babies
+Inpatient mitigators:
+ - ambulatory emergency care: change the way we extract to use a more stable and workable code list
+ - evidence based interventions: update code lists
```
[the remark this answers is not in the document]
Comment (StatsRhian) [reply data_extraction/data_log.qmd:20]: @tomjemmett any specific bits you want to highlight here? Could be lines of code, or commits.

**Review by yiwen-h — APPROVED** [review]
Happy to approve but just want final final confirmation from @tomjemmett before merge!

**Review by tomjemmett — CHANGES_REQUESTED** [review]
mostly looks fine, we are missing:

Inpatients
- alcohol_partially_attributable: errors in the old code list meant some codes were missed, and some codes were not properly distinguishing between the mortality/morbidity cases
- excess_beddays: an error in the way we handled the old csv meant some NA values were treated as 0, flagging most activity as being an excess bedday

File: data_extraction/data_log.qmd [hunk 3]
```diff
@@ -5,6 +5,30 @@ order: 250
...
+Date updated: 13/11/2024
+We have migrated our data pre-processing pipeline from SQL to Databricks. The new data pre-processing scripts are available in the public repository [nhp_data](https://github.com/The-Strategy-Unit/nhp_data). This repository supersedes the [nhp_sql](https://github.com/The-Strategy-Unit/nhp_sql) repositiory which will be publically archived and no longer maintained.
+
+As part of migrating our data pre-processing, there are a number of changes to the data.
+
+Inpatient data:
+  - Include records where `patientid` is `NULL`
+  - Include readmission within 28 days for well babies
+  - Fixed bug in `has_procedure` which was failing to filter codes beginning with `U`, `Y` or `Z`
+Inpatient mitigators:
```
Comment (tomjemmett) [inline data_extraction/data_log.qmd:19]:
```suggestion
  - Added a new flag `maternity_delivery_in_spell`, which looks to see if any episode in the spell had `maternity_episode_type=1`, the same [logic used to create the official published statistics on delivery episodes](https://digital.nhs.uk/data-and-information/publications/statistical/nhs-maternity-statistics) (applied at spell end level)
Inpatient mitigators:
```

**Review by tomjemmett — CHANGES_REQUESTED** [review]

File: data_extraction/data_log.qmd [hunk 4]
```diff
@@ -5,6 +5,30 @@ order: 250
 
 This page contains information about changes to the data underpinning the NHP model. If there are no changes logged for a particular version (such as 1.0) then the changes were only to the model code, and not to the underlying data.
 
+## Version 3.0
+
+Date updated: 13/11/2024
+We have migrated our data pre-processing pipeline from SQL to Databricks. The new data pre-processing scripts are available in the public repository [nhp_data](https://github.com/The-Strategy-Unit/nhp_data). This repository supersedes the [nhp_sql](https://github.com/The-Strategy-Unit/nhp_sql) repositiory which will be publically archived and no longer maintained.
```
[the remark this answers is not in the document]
Comment (yiwen-h) [reply data_extraction/data_log.qmd:11]: Nah I think don't bother

[the remark this answers is not in the document]
Comment (yiwen-h) [reply data_extraction/data_log.qmd:11]: Plan is just to archive and state there are known bugs I think

**Review by tomjemmett — APPROVED** [review]
lgtm

**Pull request merged**